packages/core/test/invoke.test.ts

```diff
@@ -69,7 +70,7 @@ const fetcherMachine = Machine({
       invoke: {
         src: fetchMachine,
         data: {
-          userId: (ctx) => ctx.selectedUserId
+          userId: (ctx: any) => ctx.selectedUserId
```

Comment:
places like this can't be typed in any better way because the `PropertyMapper` type is used here:
https://github.com/statelyai/xstate/blob/59fc815b288537e8ac1cf700a58ac541a95627fa/packages/core/src/types.ts#L366
and `PropertyMapper` looks like this:
https://github.com/statelyai/xstate/blob/59fc815b288537e8ac1cf700a58ac541a95627fa/packages/core/src/types.ts#L1319-L1327

The problem is that in here `TParams` are declared using `any` because we can't currently "source" the true type from anywhere. And since `PropertyMapper` includes a bit like this:
```
| TParams[K];
```
where `TParams` is effectively just `any`. And this `any` just downgrades the whole union to `any` because `T | any` is just `any`.

So TS actually can't provide a type for `ctx` here, even though the second member of the union could be used to infer this because that other member of the union makes it impossible.